org.osgi.test.junit5/src/test/java/org/osgi/test/junit5/context/BundleContextExtension_BundleContextInjectionTest.java

```diff
@@ -0,0 +1,142 @@
+package org.osgi.test.junit5.context;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.test.common.annotation.InjectBundleContext;
+import org.osgi.test.junit5.testutils.OSGiSoftAssertions;
+
+@ExtendWith(BundleContextExtension.class)
+public class BundleContextExtension_BundleContextInjectionTest {
+
+	static final Bundle			bundle	= FrameworkUtil.getBundle(BundleContextExtension_BundleContextInjectionTest.class);
+
+	@InjectBundleContext
+	static BundleContext		staticBC;
+
+	@InjectBundleContext
+	BundleContext				bundleContext;
+
+	static OSGiSoftAssertions	staticSoftly;
+
+	OSGiSoftAssertions softly;
+
+	@BeforeAll
+	static void beforeAll(@InjectBundleContext BundleContext bc) {
+		staticSoftly = new OSGiSoftAssertions();
+		staticSoftly.assertThat(staticBC)
+			.as("staticBC:beforeAll")
+			.isNotNull()
+			.isSameAs(bc)
+			.refersToBundle(bundle);
+		staticSoftly.assertAll();
+	}
+
+	@BeforeEach
+	void beforeEach(@InjectBundleContext BundleContext bc) {
+		softly = new OSGiSoftAssertions();
+		softly.assertThat(bundleContext)
+			.as("bundleContext:beforeEach")
+			.isNotNull()
+			.isSameAs(bc)
+			.refersToBundle(bundle);
+		softly.assertAll();
+	}
+
+	@Test
+	void innerTest(@InjectBundleContext BundleContext bc) {
+		softly = new OSGiSoftAssertions();
+		softly.assertThat(bundleContext)
+			.as("bundleContext:innerTest")
+			.isNotNull()
+			.isSameAs(bc)
+			.refersToBundle(bundle);
+		softly.assertAll();
+	}
+
+	@ParameterizedTest
+	@ValueSource(ints = {
+		1, 2, 3
+	})
+	void parameterizedTest(int parameter, @InjectBundleContext BundleContext bc) {
```

Comment:
more unused parameters. what's the goal here?

Reply:
As noted above, the goal is to test that the extension works with different types of test nodes. Parameterized tests are one of the different types of test node, and you can't have a parameterized test without parameters.

Reply:
comments?